firedrake/slate/static_condensation/scpc.py

```diff
@@ -68,7 +71,7 @@ def initialize(self, pc):
         for bc in cxt_bcs:
             if bc.function_space().index != c_field:
                 raise NotImplementedError("Strong BC set on unsupported space")
-            bcs.append(DirichletBC(Vc, bc.function_arg, bc.sub_domain))
+            bcs.append(DirichletBC(Vc, 0, bc.sub_domain))
```

Comment:
Do we really want to set this to 0? Or did you only change it for debugging purposes?

Reply:
We need to apply homogeneous BCs on the condensed residual regardless of what the inhomogenous values are.
Applying BCs to a matrix is also independent of the inhomogeneity.
The inhomogeneity is always taken care of by the outermost linear solver i.e. `LinearVariationalSolver` by modifying the original RHS.

If the preconditioner keeps track of the inhomogeneity and does something with it, such as applying these values to the solution, then it would no longer be a linear transformation, but an affine one. By homogenizing BCs within the PC, we prevent this sort of errors.

In (not so) old firedrake `assemble()` used to have the kwarg `assembly_type=residual`, that would just apply homogeneous BCs, this functionality is now hidden from the user. Nevertheless, this is can be achieved with `OneFormAssembler` via `zero_bc_nodes=True`.

Reply:
OK, I think I understand now

Reply:
Mhhh okay I will look into it.